adam/language_specific/english/english_language_generator.py

```diff
@@ -193,7 +194,34 @@ def _translate_object_to_noun(
             if (dependency_node.part_of_speech != PROPER_NOUN) and (
                 MASS_NOUN not in lexicon_entry.properties
             ):
-                if count == 1:
+
+                possession_relations = [
+                    relation
+                    for relation in self.situation.relations
+                    if relation.relation_type == HAS and relation.second_slot == _object
+                ]
+                if len(possession_relations) > 1:
+                    raise RuntimeError("Can not handle multiple possession relations")
+                elif len(possession_relations) == 1:
+                    possessor = possession_relations[0].first_slot
+                    # If speaker possesses the noun
+                    if IS_SPEAKER in possessor.properties:
+                        quantifier_node = DependencyTreeToken("my", DETERMINER)
+                        quantifier_role = DETERMINER_ROLE
+                    # If addressee possess the noun
+                    elif IS_ADDRESSEE in possessor.properties:
+                        quantifier_node = DependencyTreeToken("your", DETERMINER)
+                        quantifier_role = DETERMINER_ROLE
+                    # otherwise, if the agent possess the noun
+                    elif possessor == only(
```

Comment:
Actually, I think the desired behavior is to use _'s_ if a non-agent is the possessor; if the agent is a possessor, just fall back on the regular determiner behavior.  

Also this is getting complex enough we should pull it out into its own `add_determiner` method.